packages/core/admin/ee/server/services/review-workflows/stages.js

```diff
@@ -61,9 +62,36 @@ module.exports = ({ strapi }) => {
       return strapi.db.transaction(async () => {
         const newStages = await this.createMany(created, { fields: ['id'] });
         const stagesIds = stages.map((stage) => stage.id ?? newStages.shift().id);
+        const defaultWorkflow = await getDefaultWorkflow({ strapi });
 
         await mapAsync(updated, (stage) => this.update(stage.id, stage));
-        await mapAsync(deleted, (stage) => this.delete(stage.id));
+        await mapAsync(deleted, async (stage) => {
+          // Find any entities related to this stage
+          const stageInfo = await this.findById(stage.id, {
+            workflowId: defaultWorkflow.id,
+            populate: ['related'],
+          });
```

Comment:
I do think this is the best approach, still maybe in a blitz session it's worth to test for performance.

To check what happens if we delete a stage related to 100k entities? (Seems unlikely to happen, but maybe worth to know what happens 🤷‍♂️ )

Reply:
can we paginate the population of related? 🤔

Reply:
It is possible if we use `entityService.load` , it can be used to load  relational fields  of an entity.

Reply:
`entityService.load(uid, id, ['related'], {
  start: 10,
  limit: 15,
})`

This should theoretically paginate the `related` field from the `stage`

Reply:
I'm not sure I follow, why would we paginate related?

Reply:
If we want to handle large data amounts, we need to paginate everything that can be infinite (like the number of related entities). 👍